auth/src/user.rs

```diff
@@ -0,0 +1,287 @@
+use std::{fmt::Formatter, str::FromStr};
+
+use async_trait::async_trait;
+use axum::{
+    extract::{FromRef, FromRequestParts},
+    headers::{authorization::Bearer, Authorization},
+    http::request::Parts,
+    TypedHeader,
+};
+use rand::distributions::{Alphanumeric, DistString};
+use serde::{Deserialize, Deserializer, Serialize};
+use sqlx::{query, Pool, Row, Sqlite};
+use tracing::{trace, Span};
+
+use crate::{api::RouterState, error::Error};
+
+#[async_trait]
+pub(crate) trait UserManagement {
+    async fn create_user(&self, name: AccountName) -> Result<User, Error>;
```

Comment:
Maybe we should add the role to this interface?

Reply:
I did it with path params, a bit unsure if it should have been with json instead..

Reply:
We could also have gone with Post (form) data. But this will be fine for now, thanks